phpBB/phpbb/plupload/plupload.php

```diff
@@ -283,15 +283,51 @@ public function generate_resize_string()
 	*/
 	public function get_chunk_size()
 	{
-		$max = min(
-			$this->php_ini->getBytes('upload_max_filesize'),
-			$this->php_ini->getBytes('post_max_size'),
-			max(1, $this->php_ini->getBytes('memory_limit')),
-			$this->config['max_filesize']
-		);
+		$max = 0;
+
+		$limit = $this->php_ini->getBytes('memory_limit');
+
+		// unlimited is -1 for memory_limit. 0 would be an invalid configuration.
+
+		if ($limit > 0)
+		{
+			$max = $limit;
+		}
+
+		// For all remaining limits, 0 means "unlimited".
+
+		// For each limit, if there is a non-unlimited value to
+		// apply, apply the limit if it's less than whatever non-
+		// unlimited max value is currently set.  Also, apply the
+		// limit if the current max value is otherwise unlimited.
+
+		$limit = $this->php_ini->getBytes('upload_max_filesize');
+
+		if ($limit > 0)
+		{
+			$max = min($limit, max($max, $limit));
+		}
+
+		$limit = $this->php_ini->getBytes('post_max_size');
+
+		if ($limit > 0)
+		{
+			$max = min($limit, max($max, $limit));
+		}
+
+		$limit = $this->config['max_filesize'];
+
+		if ($limit > 0)
+		{
+			$max = min($limit, max($max, $limit));
+		}
```

Comment:
No way to beautify this code?

Reply:
This code looks good for me, as simple as possible...
It's also noob friendly... & Readable...